src/rtt.cpp

```diff
@@ -0,0 +1,590 @@
+        {
+            baton->lowlevelError = loadHighlevelStatus;
+            return RTTCouldNotOpenHighlevelLibrary;
+        }
+
+        const nrfjprogdll_err_t openHighlevelStatus = highLevelFunctions.dll_open(nullptr, nullptr, nullptr);
+
+        if (openHighlevelStatus != SUCCESS)
+        {
+            releaseHighLevel();
+            baton->lowlevelError = openHighlevelStatus;
+            return RTTCouldNotOpenHighlevelLibrary;
+        }
+
+        Probe_handle_t probe;
+        const nrfjprogdll_err_t initProbeStatus = highLevelFunctions.probe_init(&probe, baton->serialNumber, nullptr);
+
+        if (initProbeStatus != SUCCESS)
+        {
+            highLevelFunctions.dll_close();
+            releaseHighLevel();
+            baton->lowlevelError = initProbeStatus;
+            return RTTCouldNotGetDeviceInformation;
+        }
+
+        probe_info_t probeInfo;
+        device_info_t deviceInfo;
+        library_info_t libraryInfo;
+
+        RETURN_ERROR_ON_FAIL(highLevelFunctions.get_probe_info(probe, &probeInfo), RTTCouldNotGetDeviceInformation);
+        RETURN_ERROR_ON_FAIL(highLevelFunctions.get_device_info(probe, &deviceInfo), RTTCouldNotGetDeviceInformation);
+        RETURN_ERROR_ON_FAIL(highLevelFunctions.get_library_info(probe, &libraryInfo), RTTCouldNotGetDeviceInformation);
+
+        RETURN_ERROR_ON_FAIL(highLevelFunctions.reset(probe, RESET_SYSTEM), RTTCouldNotGetDeviceInformation);
+        RETURN_ERROR_ON_FAIL(highLevelFunctions.probe_uninit(&probe), RTTCouldNotGetDeviceInformation);
+        highLevelFunctions.dll_close();
+
+        releaseHighLevel();
+
+        uint32_t clockSpeed = probeInfo.clockspeed_khz;
+        device_family_t family = deviceInfo.device_family;
+        std::string jlinkarmlocation = libraryInfo.file_path;
+
+        RETURN_ERROR_ON_FAIL((nrfjprogdll_err_t)loadnRFjprogFunctions(&dll_function), RTTCouldNotLoadnRFjprogLibrary);
+
+        libraryLoaded = true;
+
+        RETURN_ERROR_ON_FAIL(dll_function.open_dll(jlinkarmlocation.c_str(), &RTT::logCallback, family), RTTCouldNotOpennRFjprogLibrary);
+
+        RETURN_ERROR_ON_FAIL(dll_function.connect_to_emu_with_snr(baton->serialNumber, clockSpeed), RTTCouldNotConnectToDevice);
+        RETURN_ERROR_ON_FAIL(dll_function.connect_to_device(), RTTCouldNotConnectToDevice);
+
+        if (baton->hasControlBlockLocation) {
+            RETURN_ERROR_ON_FAIL(dll_function.rtt_set_control_block_address(baton->controlBlockLocation), RTTCouldNotFindControlBlock);
+        }
+
+        rttStartTime = std::chrono::high_resolution_clock::now();
+        RETURN_ERROR_ON_FAIL(dll_function.rtt_start(), RTTCouldNotStartRTT);
+
+
+        while (true) {
+            bool controlBlockFound = false;
+            RETURN_ERROR_ON_FAIL(dll_function.rtt_is_control_block_found(&controlBlockFound), RTTCouldNotCallFunction);
+            std::chrono::high_resolution_clock::time_point attemptedStartupTime = std::chrono::high_resolution_clock::now();
```

Comment:
Only used if controlBlockFound is false. Move it after break condition along with a sensible sleep.